File: 0728-self-dividing-numbers/0728-self-dividing-numbers.py
class Solution:
    def helper(self, n: int) -> bool:
        if n % 10 == 0:
            return False
        original = n
        while n:
            digit = n % 10
            if digit == 0 or original % digit != 0:
                return False
            n //= 10
        return True
    
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        res = []
        for i in range(left, right+1):
            if self.helper(i):
                res.append(i)
        return res

File: 0728-self-dividing-numbers/0728-self-dividing-numbers.kt
class Solution {
    fun helper(n: Int): Boolean {
        val original = n
        var num = n
        while (num !== 0) {
            val digit = num % 10
            if (digit === 0 || original % digit !== 0) {
                return false
            }
            num /= 10
        }
        return true
    }
    
    fun selfDividingNumbers(left: Int, right: Int): List<Int> {
        val res = ArrayList<Int>()
        for (i in left..right) {
            if (this.helper(i)) {
                res += i
            }
        }
        return res
    }
}